carbon-design-system/carbon-addons-iot-react
# refactor(ValueCard): new design and precision fixes

Internal issue: https://github.ibm.com/wiotp/monitoring-dashboard/issues/1694
Closes #1892 
Closes #1141 
Part of #370 
Closes #1276 

**Summary**

*Note*: I would recommend not relying on the diff for the ValueCard files for this review as this is essentially a full rewrite. It will be much easier to read the files without the diff to see the new logic and review as if this was a new component being introduced.

- Redesign primarily to handle truncation and wrapping differently when precision is set
- Required a very large refactor as the old logic had very dense logic and conditionals around all of the different card sizes for truncating and wrapping

**Change List (commits, features, bugs, etc)**

- Removed styled components and swapped the dynamic variables for scss variables
- Removed `formatChartNumberWithPrecision` in favor of `formatNumberWithPrecision` as they had the same exact logic
- Refactored `formatNumberWithPrecision` to instead use the native I18n class to format numbers based on locale (may need to check other places in the code that we're using external libraries or our own logic to try and determine the locale format
- Extracted all attribute styling logic to Attribute to keep separation of concerns
- Renamed, removed, and refactored some of the stories to better reflect the possible use-cases

**Acceptance Test (how to verify the PR)**

1. ValueCard
  - ensure that each ValueCard story adheres to the new design spec seen in the internal issue above
  - Try using the font size knob to see that the font size responds correctly for the value
  - Try using the isNumberValueCompact knob and change the locale to see that it translates correctly

2. DashboardEditor
  - Add a new ValueCard, go to the settings tab, and try the font size number input and abbreviate numbers input. You should see both getting updated in the actions
